src/utils/guildRole.ts

```diff
@@ -84,3 +85,40 @@ export async function removeGuildRole(details: memberGroupRole, env: env) {
     return INTERNAL_SERVER_ERROR;
   }
 }
+
+export async function getGuildRoles(
+  env: env
+): Promise<Array<GuildRole> | undefined> {
+  const guildDetailsUrl = `${DISCORD_BASE_URL}/guilds/${env.DISCORD_GUILD_ID}`;
+
+  try {
+    const response = await fetch(guildDetailsUrl, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+    });
+
+    if (!response.ok) {
+      return undefined;
```

Comment:
Nit: you can just return here. Undefined in explicit

Reply:
But I believe this is the case when the response is not ok, so an appropriate error should be thrown instead of returning undefined.